tests/scalability/generate_brac_data/generate-brac-data.js

```diff
@@ -0,0 +1,221 @@
+    createDataDoc(reportsDirectory, assesmentSurvey._id + dataExtension, JSON.stringify(assesmentSurvey, {}, 2));
+    const assesmentFollowUpSurvey = bracSurvey.generateBracSurvey(
+      'assesment_follow_up', directParentPlace, place, person);
+    createDataDoc(reportsDirectory, assesmentFollowUpSurvey._id + dataExtension,
+      JSON.stringify(assesmentFollowUpSurvey, {}, 2));
+  }
+};
+
+const generateHierarchy = (type, placeName, numberOfPersons,
+  parentFirstLevelId, parentSecondLevelId, directParentPlace) => {
+  const place = bracPlaceFactory.generateBracPlace(placeName, type,
+    setParents(parentFirstLevelId, parentSecondLevelId));
+  let parentPersonThirdLevelId = null;
+  let parentPersonSecondLevelId = null;
+  let parentPersonFirstLevelId = null;
+  if (type === 'district_hospital') {
+    parentPersonFirstLevelId = place._id;
+  }
+  if (type === 'health_center') {
+    parentPersonFirstLevelId = place._id;
+    parentPersonSecondLevelId = parentFirstLevelId;
+    const randomIndex = Faker.faker.datatype.number(numberOfManagersPerDistrictHospital - 1);
+    place.supervisor = managers[randomIndex]._id;
+  }
+  if (type === 'clinic') {
+    parentPersonFirstLevelId = place._id;
+    parentPersonSecondLevelId = parentFirstLevelId;
+    parentPersonThirdLevelId = parentSecondLevelId;
+  }
+  let isPrimaryContact = true;
+  for (let i = 0; i < numberOfPersons; i++) {
+    const person = generatePerson(type, setParents(parentPersonFirstLevelId,
+      parentPersonSecondLevelId, parentPersonThirdLevelId), isPrimaryContact);
+    if (isPrimaryContact) {
+      place.contact = person;
+      createDataDoc(preconditionDataDirectory, place._id + dataExtension, JSON.stringify(place, {}, 2));
+    }
+    const needUser = pairPlaceTypesNeedsUsers[type];
+    if (needUser) {
+      generateUser(type, place._id, person.short_name.toLowerCase() + placeName + 'user' + i, person, isPrimaryContact);
+    }
+    if (bracPersonFactory.shouldGenerateSurvey(person)) {
+      generateReports(directParentPlace, place, person, (i % 2 !== 0));
+    }
+    isPrimaryContact = false;
+  }
+  return place;
+};
+
+const generateData = () => {
+  createDataDirectory(dataDirectory, preconditionDirectory);
+  createDataDirectory(dataDirectory, mainDirectory);
+  createDataDirectory(path.join(dataDirectory, preconditionDirectory), jsonDirectory);
+  for (let dh = 0; dh < numberOfDistrictHospitals; dh++) {
+    managers.splice(0, managers.length);
+    const districtHospital = generateHierarchy(
+      'district_hospital',
+      districtHospitalName + 'districthospital' + dh,
+      numberOfManagersPerDistrictHospital);
+
+    for (let hc = 0; hc < numberOfHealthCentersPerDistrictHospital; hc++) {
+      const healthCenter = generateHierarchy(
+        'health_center',
+        districtHospitalName + 'districthospital' + dh + 'healthcenter' + hc,
+        numberOfChwPerHealthCenter, districtHospital._id);
```

Comment:
This argument list is really long. Can you please define the name of the place separately?  

```suggestion
const healthCenterName = districtHospitalName + 'districthospital' + dh + 'healthcenter' + hc;
const healthCenter = await generateHierarchy('health_center', healthCenterName, numberOfChwPerHealthCenter, districtHospital._id);
```